File: python/Problem003.py
#
# This file contains a solution to the problem number 3 of the Project Euler, last modified on 5/3/2018.
# Copyright (c) Matthieu Le Boucher. All rights reserved.
#

def result():
    n = 600_851_475_143
    currentFactor = 2

    while currentFactor * currentFactor < n:
        while n % currentFactor == 0:
            n = n / currentFactor

        currentFactor = currentFactor + 1

    return n

if __name__ == "__main__":
    print(result())

File: cpp/Problem003.cpp
#include "EulerSolution.h"

/*
 * @author Matthieu Le Boucher <matt.leboucher@gmail.com>
 */

class Problem003 : public EulerSolution {
public:
    int number() override {
        return 3;
    }

    double solution() override {
        // Target number is too large for 32-bit integer representation.
        long n = 600851475143L;

        // Look for prime factors in [2, sqrt(n)].
        int currentFactor = 2;
        while (currentFactor * currentFactor < n) {
            while (n % currentFactor == 0)
                n /= currentFactor;

            currentFactor++;
        }

        return n;
    }

    double answer() override {
        return 6857;
    }
};